File: mcp-server/lua/get-trade-routes.lua
-- Get trade route info for AI accessibility
-- Returns active routes with yields
-- Addresses trade route gap identified in Issue #469

local pPlayer = Players[playerID]
if not pPlayer then
  return nil
end

local result = {
  -- Summary
  TotalRoutes = pPlayer:GetNumInternationalTradeRoutesUsed(),
  AvailableSlots = pPlayer:GetNumInternationalTradeRoutesAvailable() - pPlayer:GetNumInternationalTradeRoutesUsed(),
  MaxRoutes = pPlayer:GetNumInternationalTradeRoutesAvailable(),

  -- Active routes
  ActiveRoutes = {},

  -- Idle trade units (can establish new routes)
  IdleTradeUnits = {},
}

-- Get all trade routes
local tradeRoutes = pPlayer:GetTradeRoutes()
for i, route in ipairs(tradeRoutes) do
  local fromCity = pPlayer:GetCityByID(route.FromID)
  local toPlayer = Players[route.ToPlayerID]
  local toCity = toPlayer and toPlayer:GetCityByID(route.ToID) or nil

  if fromCity and toCity then
    local routeData = {
      Domain = route.Domain == GameInfoTypes.DOMAIN_SEA and "Sea" or "Land",
      FromCity = fromCity:GetName(),
      ToCity = toCity:GetName(),
      ToCiv = toPlayer:GetCivilizationShortDescription(),

      TurnsRemaining = route.TurnsLeft,

      -- Is it internal?
      IsInternal = route.ToPlayerID == playerID,
    }

    -- Get yields
    -- Gold yields
    routeData.FromGold = route.FromGPT or 0
    routeData.ToGold = route.ToGPT or 0

    -- Food/Production (internal routes)
    if routeData.IsInternal then
      routeData.FromFood = route.FromFood or 0
      routeData.FromProduction = route.FromProduction or 0
    end

    -- Science from tech difference
    routeData.FromScience = route.FromScience or 0
    routeData.ToScience = route.ToScience or 0

    -- Religious pressure
    routeData.FromReligiousPressure = route.FromReligiousPressure or 0
    routeData.ToReligiousPressure = route.ToReligiousPressure or 0

    table.insert(result.ActiveRoutes, routeData)
  end
end

-- Find idle trade units
for pUnit in pPlayer:Units() do
  local unitInfo = GameInfo.Units[pUnit:GetUnitType()]
  if unitInfo and unitInfo.Trade then
    -- Check if this unit has moves and isn't on a route
    if pUnit:GetMoves() > 0 and not pUnit:IsTrade() then
      local plot = pUnit:GetPlot()
      local city = plot and plot:GetPlotCity() or nil

      table.insert(result.IdleTradeUnits, {
        ID = pUnit:GetID(),
        Name = Locale.ConvertTextKey(unitInfo.Description),
        X = pUnit:GetX(),
        Y = pUnit:GetY(),
        InCity = city and city:GetName() or nil,
        Domain = unitInfo.Domain == "DOMAIN_SEA" and "Sea" or "Land",
      })
    end
  end
end

return result

File: mcp-server/src/tools/knowledge/get-trade-routes.ts
/**
 * Trade routes tool for AI accessibility (Issue #469)
 */

import { ToolBase } from "../base.js";
import * as z from "zod";
import { ToolAnnotations } from "@modelcontextprotocol/sdk/types.js";
import { getTradeRoutes } from "../../knowledge/getters/trade-routes.js";
import { MaxMajorCivs } from "../../knowledge/schema/base.js";

const GetTradeRoutesInputSchema = z.object({
  PlayerID: z.number().min(0).max(MaxMajorCivs - 1).optional()
    .describe("Player ID (default: 0)"),
});

const TradeRouteSchema = z.object({
  Domain: z.enum(["Land", "Sea"]),
  FromCity: z.string(),
  ToCity: z.string(),
  ToCiv: z.string(),
  TurnsRemaining: z.number(),
  IsInternal: z.boolean(),

  FromGold: z.number(),
  ToGold: z.number(),
  FromFood: z.number().optional(),
  FromProduction: z.number().optional(),
  FromScience: z.number().optional(),
  ToScience: z.number().optional(),
  FromReligiousPressure: z.number().optional(),
  ToReligiousPressure: z.number().optional(),
});

const IdleTradeUnitSchema = z.object({
  ID: z.number(),
  Name: z.string(),
  X: z.number(),
  Y: z.number(),
  InCity: z.string().optional(),
  Domain: z.enum(["Land", "Sea"]),
});

const GetTradeRoutesOutputSchema = z.object({
  TotalRoutes: z.number(),
  AvailableSlots: z.number(),
  MaxRoutes: z.number(),
  ActiveRoutes: z.array(TradeRouteSchema),
  IdleTradeUnits: z.array(IdleTradeUnitSchema).describe("Trade units that can establish new routes"),
});

class GetTradeRoutesTool extends ToolBase {
  readonly name = "get-trade-routes";

  readonly description = `Get all trade route information with yields.

Shows:
- Active routes with gold, food, production, science yields
- Turns remaining on each route
- Internal vs international routes
- Religious pressure from routes
- Available trade route slots
- Idle trade units that can establish new routes`;

  readonly inputSchema = GetTradeRoutesInputSchema;
  readonly outputSchema = GetTradeRoutesOutputSchema;

  readonly annotations: ToolAnnotations = { readOnlyHint: true };

  async execute(args: z.infer<typeof this.inputSchema>): Promise<z.infer<typeof this.outputSchema>> {
    const playerID = args.PlayerID ?? 0;
    const data = await getTradeRoutes(playerID);

    if (!data) {
      throw new Error('Failed to get trade routes data');
    }

    return data;
  }
}

export default function createGetTradeRoutesTool() {
  return new GetTradeRoutesTool();
}
